azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VirtualMachineApiLiveTest.java

```diff
@@ -133,18 +135,39 @@ public void testGetInstanceView() {
       assertTrue(!vmi.statuses().isEmpty());
    }
 
-   @Test(dependsOnMethods = "testStart")
-   public void testStop() {
-      api().stop(vmName);
-      assertTrue(stateReached(PowerState.STOPPED), "stop operation did not complete in the configured timeout");
-   }
-
    @Test(dependsOnMethods = "testGet")
    public void testStart() {
       api().start(vmName);
       assertTrue(stateReached(PowerState.RUNNING), "start operation did not complete in the configured timeout");
    }
 
+   @Test(dependsOnMethods = "testStart")
+   public void testUpdate() {
+      VirtualMachine vm = api().get(vmName);
+      VirtualMachineProperties oldProperties = vm.properties();
+      StorageProfile oldStorageProfile = oldProperties.storageProfile();
+
+      String blob = storageService.storageServiceProperties().primaryEndpoints().get("blob");
+      VHD vhd = VHD.create(blob + "vhds/" + vmName + "new-data-disk.vhd");
+      DataDisk newDataDisk = DataDisk.create(vmName + "new-data-disk", "1", 1, vhd, "Empty");
+      List<DataDisk> oldDataDisks = oldStorageProfile.dataDisks();
+
+      ImmutableList<DataDisk> newDataDisks = ImmutableList.<DataDisk> builder().addAll(oldDataDisks).add(newDataDisk).build();
+      StorageProfile newStorageProfile = oldStorageProfile.toBuilder().dataDisks(newDataDisks).build();
+      VirtualMachineProperties newProperties = oldProperties.toBuilder().storageProfile(newStorageProfile).build();
+
+      VirtualMachine newVm = vm.toBuilder().properties(newProperties).build();
+      vm = api().createOrUpdate(vmName, newVm.location(), newVm.properties(), newVm.tags(), newVm.plan());
+
+      assertEquals(vm.properties().storageProfile().dataDisks().size(), 2);
```

Comment:
Perhaps assert it's equal to `oldDataDisks.size() + 1` (or assert earlier that the oldDataDisks size is 1)?